### DISCUSSION TITLE
Accessing Historical Alarms and Events Over OPC A&E Server using Python / .NET / Java or any other platform

### DISCUSSION THREAD

**Discussion by rajnishnationfirst** [FreeOpcUa/python-opcua#1230] [Q&A]
Hi ,

I am implementing and want to capture Historical Alarms and Events over OPC A&E Server ( Not OPC UA or OPC HDA ) created in Siemens WinCC SCADA.

I tried HDA data over OPC UA and implemented in Python Free OPCUA SDK and working fine. But the WinCC does not support the capturing Historical Alarms over OPC UA and yet the OPC UA has node.read_events_history(start_time,end_time ,0) , but i cannot use this function on OPCA&E server available in python .

The below is the snap of WinCC Siemens SCADA OPC A&E Server and don't have node Ids and Tags configuration.

<img width="792" alt="alarms" src="https://user-images.githubusercontent.com/64892368/105334365-caf6f300-5bfc-11eb-9498-bef24ec68430.PNG">

Could any one help with some open source SDK's available in Python , .NET or Java or Java Script for OPC A&E Server capturing historical alarms and events. ?

Thanks and Regards

Rajnish Vishwakarma

**Comment by AndreasHeine** [comment]
the problem with you question is that this repository is only for "OPC Unified Architecture", "OPC A&E" is part of the older "OPC Classic" (COM/DCOM) based standard!

**Comment by rajnishnationfirst** [comment]
Thanks Andreas , 

How can get historical alarms over OPC A&E server or from which server .?

Thanks

**Reply by AndreasHeine** [reply]
> How can get historical alarms over OPC A&E server or from which server .?

i dont understand the question! this is **OPC UA** not OPC A&E (completly different transport protocols), you should search for such COM/DCOM libs in python but as far as i know its bound to windows and is kind of  insecure. there are gatewaysolutions out there (Kepware, Prosys, ...) which can be used to get the data, but this is up to you!

**Reply by harriv** [reply]
There are products like this to bridge classic OPC and OPC UA: https://prosysopc.com/products/opc-ua-gateway/

**Comment by duduyoyo** [comment]
Check [this](https://github.com/duduyoyo/WebSocket4OPC) out, you can use Python/JavaScript/C#/Java etc. to access Classic OPC DA/HDA/AE servers. Cheers!
